var/spack/repos/builtin/packages/environment-modules/package.py

```diff
@@ -27,13 +27,13 @@
 
 
 class EnvironmentModules(Package):
-    """The Environment Modules package provides for the dynamic
-    modification of a user's environment via modulefiles."""
+    """The Environment Modules package provides for the dynamic modification of
+    a user's environment via modulefiles."""
 
-    homepage = "https://sourceforge.net/p/modules/wiki/Home/"
-    url = "http://prdownloads.sourceforge.net/modules/modules-3.2.10.tar.gz"
+    homepage = "https://sourceforge.net/p/modules/"
+    url = "http://prdownloads.sourceforge.net/modules/modules-4.1.4.tar.gz"
 
-    version('3.2.10', '8b097fdcb90c514d7540bb55a3cb90fb')
+    version('4.1.4', sha256='965b6056ea6b72b87d9352d4c1db1d7a7f9f358b9408df2689d823b932d6aa7f')
 
```

Comment:
We generally like to keep old versions in the package (see also above regarding backwards compatibility).

Reply:
Modules version 3 is not compatible with fish, that's why I updated it.

Reply:
By the way: when you say you want to keep old versions, do you mean "as well" (i.e. spack should provide a choice of versions)? Or are you generally opposed to upgrading any packages

Reply:
It is always an "as well", spack will default to the newest version, but it should retain the information to build old versions, as one of the main points of spack is reproducibility

Reply:
Each installed spec (i.e. each configuration of versions and variants of a package) will also produce a module. There is quite a lot of infrastructure to modify these, but you should ask @alalazo he knows this part inside out. Or maybe talk to us on slack if you need/want help